### PR TITLE
Contributing and checking commits

### LINKED ISSUE #6: provide a documentation contribution guidelines doc
Provide a contributions guideline doc that informs new users how to contribute to the project, licensing permitted, commit message style etc

### LINKED ISSUE #10: Add code scanning
https://docs.github.com/en/code-security/code-scanning/introduction-to-code-scanning/about-code-scanning-with-codeql

Please enable automated github vulnerability scans at the very least.

### PR DESCRIPTION
**Pull request by StaticRocket**
Add a short file detailing how one can contribute to the project.

Add a commit check for PRs to make sure they are following the rules outlined in the CONTRIBUTING document.

Resolves #6 
Resolves #10

### CODE REVIEW

**Comment by StaticRocket** [comment]
@nmenon

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+documentation for a platform that matches the ifconfig prior to submitting a PR.
+
+## Licensing
+
+Scripts and other code specific additions should be licensed MIT, with the below
+header and copyright notice placed in a comment near the top of the file.
+
+```text
+SPDX-License-Identifier: MIT
+Copyright (C) 2024 Texas Instruments Incorporated - https://www.ti.com
```
Comment (nmenon) [inline CONTRIBUTING.md:114]: This makes sense only for contributors who are from TI. We just need to insist on MIT license for redistribution. If other devs contribute using their personal copyrights, we should be fine. We do not need copyright attribution to TI.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+```
+
+All other files inherit the generic CC-BY-SA-4.0 license and do not require an
+explicit license / copyright notice.
+
+## Fighting workflows
+
+The workflows are here to help, but occasionally they can raise false positives.
+In such an event, please ping a reviewer and explain why the failure should be
```
Comment (nmenon) [inline CONTRIBUTING.md:123]: Explain an example, please.

**Review by nmenon — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+Not all parsers will respect this directive, so be sure to try compiling the
+documentation for a platform that matches the ifconfig prior to submitting a PR.
+
+## Licensing
+
+### Community
+
+Scripts and other code specific additions are preferred to be licensed MIT.
+Other licenses are acceptable so long as they do not conflict with the existing
+MIT and CC-BY-SA-4.0 definitions.
```
Comment (nmenon) [inline CONTRIBUTING.md:113]: Just keep it simple, we are not lawyers - just MIT and CC-BY-SA-4.0 which the TI lawyers are OK with. we don't want to get into a debate at the point of a PR as to what license conflicts with which one etc..

**Review by nmenon — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+Build configuration files that hold platform specific variables are under the
+`configs/` directory.
+
+Scripts to verify source files and generally lint the documents are under the
+`bin/` directory.
+
+The container build tools are listed under the `docker/` directory.
+
+Other directories contain sphinx specific overrides that help massage the above
```
Comment (cshilwant) [inline CONTRIBUTING.md:34]:
```suggestion
Other directories contain sphinx specific overrides that helps the above
```

@StaticRocket  Seems like you are planning for a massage over the weekend 😉

**Comment by cshilwant** [comment]
Hmm, Looks like conventional commit doesn't like more than 1 colon in commit message title - https://github.com/TexasInstruments/processor-sdk-doc/actions/runs/11524739733/attempts/1#summary-32085557172

**Comment by cshilwant** [comment]
The bots looks super strict - https://github.com/StaticRocket/processor-sdk-doc/actions/runs/11526783806/attempts/1#summary-32091486204

Has a restriction in branch name as well

**Comment by nmenon** [comment]
I must say, I LOVE IT!

**Comment by StaticRocket** [comment]
Yeah, but that's more for filtering branches on the primary repo, which there shouldn't really be anything other than master unless we decide to do some massive work that requires multiple contributions.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+directive](https://www.sphinx-doc.org/en/master/usage/extensions/ifconfig.html).
+Not all parsers will respect this directive, so be sure to try compiling the
+documentation for a platform that matches the ifconfig prior to submitting a PR.
+
+## Licensing
+
+### Community
+
+Scripts and other code specific additions should be licensed MIT. Other
+modifications should be CC-BY-SA-4.0.
```
Comment (nmenon) [inline CONTRIBUTING.md:112]: I hate doing this, but let me try it:

```
Scripts and other code-specific additions should be licensed as `MIT <https://spdx.org/licenses/MIT.html>`_ using the `SPDX short identifiers <https://spdx.dev/learn/handling-license-info/>`_. Other additions or modifications should be CC-BY-SA-4.0.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+### TI Employees
+
+Scripts and other code specific additions should be licensed MIT, with the below
+header and copyright notice placed in a comment near the top of the file.
```
Comment (nmenon) [inline CONTRIBUTING.md:117]: In addition to the community guidelines, TI employees are expected to explicitly assign license to TI as in the example below:

**Comment by nmenon** [comment]
Probably not on this commit, but we should consider adding something like https://blog.alta3.com/topics/scm-and-pipelines/adding-spell-checking-as-a-github-action/ -> I wonder if github enterprise has a paid service for grammar and spell check

File: CONTRIBUTING.md (hunk 5 above)
Comment (StaticRocket) [reply CONTRIBUTING.md:112]: Ha, that's rst syntax, but I get your point.

**Review by nmenon — APPROVED** [review]

**Review by cshilwant — APPROVED** [review]

**Comment by cshilwant** [comment]
> Yeah, but that's more for filtering branches on the primary repo, which there shouldn't really be anything other than master unless we decide to do some massive work that requires multiple contributions.

I was referring to the branch names it expects from the author creating PRs as well. 
_**Branches must begin with these types: bugfix/ feature/ release/ hotfix/**_

**Comment by nmenon** [comment]
Also, please edit $subject line of PR: s/Contibuting/Contributing so that the merge message looks clean.

**Comment by StaticRocket** [comment]
> I wonder if github enterprise has a paid service for grammar and spell check

I was thinking about that, but considering all the acronyms and odd names we'd definitely need to make our own dictionary for it to be useful. That being said, if we actually did make an acronym and odd word dict, it would make for good appendix fodder.

**Review by praneethbajjuri — APPROVED** [review]

**Pull request merged**